ZPastel.Persistence/Impl/PastelRepository.cs

```diff
@@ -15,6 +16,24 @@ public PastelRepository(DataContext dataContext)
             this.dataContext = dataContext;
         }
 
+        public async Task<Page<Pastel>> Filter(PastelFilter pastelFilter)
+        {
+            var queryResults = dataContext.Set<Pastel>().AsQueryable();
+            queryResults = queryResults.Where(q => EF.Functions.Like(q.Name, $"%{pastelFilter.Name}%"));
```

Comment:
Check if pastelFilter.Name has any value. To avoid searching for an empty pastelName.

Reply:
Done